lib/compiler-singlepass/src/machine_arm64.rs

```diff
@@ -3072,38 +3075,90 @@ impl Machine for MachineARM64 {
         Ok(())
     }
     fn i32_popcnt(&mut self, loc: Location, ret: Location) -> Result<(), CompileError> {
-        let mut temps = vec![];
-
-        let src_gpr =
-            self.location_to_reg(Size::S32, loc, &mut temps, ImmType::None, true, None)?;
-        let dst_gpr =
-            self.location_to_reg(Size::S32, ret, &mut temps, ImmType::None, false, None)?;
-
-        let mut neon_temps = vec![];
-        let neon_temp = self.acquire_temp_simd().ok_or_else(|| {
-            CompileError::Codegen("singlepass cannot acquire temp gpr".to_owned())
-        })?;
-        neon_temps.push(neon_temp);
+        if self.target.is_some()
+            && self
+                .target
+                .as_ref()
+                .unwrap()
+                .cpu_features()
+                .contains(CpuFeature::NEON)
```

Comment:
This might be a bit time-consuming if the `i32_popcnt` is called a lot (which can happen), I think we can rather have this precomputed on machine creation, and simply do:
`self.supports_neon`

Thoughts @xdoardo ?

Reply:
Addressed in  [f553d9f](https://github.com/wasmerio/wasmer/pull/4526/commits/f553d9fe871e4ad091c57f423e5edc9461dc9fac)